ra2ce/analysis/adaptation/adaptation_option.py

```diff
@@ -39,71 +51,115 @@ class AdaptationOption:
+            )
+            return config_data
 
-        if not damages_section or not losses_section:
+        if (
+            not analysis_config.config_data.damages_list
+            or not analysis_config.config_data.losses_list
+        ):
             raise ValueError(
                 "Damages and losses sections are required to create an adaptation option."
             )
 
-        _damages_root = extend_path("damages", root_path)
-        _damages_section = deepcopy(damages_section)
+        # Create input for the damages analysis
+        _damages_config = deepcopy(analysis_config)
+        _damages_config.config_data = replace_paths(
+            _damages_config.config_data, "damages"
+        )
+        _damages_analysis = _damages_config.config_data.get_analysis(
+            AnalysisDamagesEnum.DAMAGES
+        )
+        _damages_input = AnalysisInputWrapper.from_input(
+            analysis=_damages_analysis,
+            analysis_config=_damages_config,
+            graph_file_hazard=analysis_config.graph_files.base_network_hazard,
+        )
 
-        _losses_root = extend_path("losses", root_path)
-        _losses_section = deepcopy(losses_section)
-        _losses_section.resilience_curves_file = extend_path(
-            "losses", losses_section.resilience_curves_file
+        # Create input for the losses analysis
+        _losses_config = deepcopy(analysis_config)
+        _losses_config.config_data = replace_paths(_losses_config.config_data, "losses")
+        _losses_analysis = _losses_config.config_data.get_analysis(
+            analysis_config.config_data.adaptation.losses_analysis
+        )
+        _losses_analysis.traffic_intensities_file = construct_path(
+            _losses_analysis.traffic_intensities_file, "losses", "input"
         )
-        _losses_section.traffic_intensities_file = extend_path(
-            "losses", losses_section.traffic_intensities_file
+        _losses_analysis.resilience_curves_file = construct_path(
+            _losses_analysis.resilience_curves_file, "losses", "input"
         )
-        _losses_section.values_of_time_file = extend_path(
-            "losses", losses_section.values_of_time_file
+        _losses_analysis.values_of_time_file = construct_path(
+            _losses_analysis.values_of_time_file, "losses", "input"
+        )
+        _losses_analysis
+        _losses_input = AnalysisInputWrapper.from_input(
+            analysis=_losses_analysis,
+            analysis_config=_losses_config,
+            graph_file=analysis_config.graph_files.base_graph,
+            graph_file_hazard=analysis_config.graph_files.base_graph_hazard,
         )
 
         return cls(
             **asdict(adaptation_option),
-            damages_root=_damages_root,
-            damages_config=_damages_section,
-            losses_root=_losses_root,
-            losses_config=_losses_section,
+            damages_input=_damages_input,
```

Comment:
I'm not entirely convinced about this method, if you need both analysis and have the original configuration, why not creating a new object that represents the tuple of them rather than having them separately and duplicating information?